mewarpx/mewarpx/poisson_pseudo_1d.py

```diff
@@ -0,0 +1,143 @@
+import numpy as np
+from scipy.sparse import csc_matrix, linalg as sla
+
+from mewarpx.mwxrun import mwxrun
+from pywarpx import callbacks
+from pywarpx.picmi import constants
+
+class PoissonSolverPseudo1D(object):
+
+    def __init__(self, left_voltage=0, right_voltage=0):
+        """Direct solver for the Poisson equation using superLU. This solver is
+        useful for pseudo 1D cases i.e. diode simulations with small x extent.
+
+        Arguments:
+            right_voltage/left_voltage (float or callable): Value or function
+            to calculate value of the potential on the left and right side of
+            the domain. If callable that function should accept simulation time
+            as an argument.
+        """
+
+        if not np.isclose(mwxrun.dx, mwxrun.dz):
+            raise RuntimeError('Direct solver requires dx = dz.')
+
+        # for historical reasons (copying from metools) the direct solver
+        # uses the transpose of the domain
+        self.nz = mwxrun.nx
+        self.nx = mwxrun.nz
+        self.dx = mwxrun.dz
+
+        self.nxguardrho = 2
+        self.nzguardrho = 2
+        self.nxguardphi = 1
+        self.nzguardphi = 1
+
+        if callable(left_voltage):
+            self.left_voltage = left_voltage
+        else:
+            self.left_voltage = lambda x: left_voltage
+
+        if callable(right_voltage):
+            self.right_voltage = right_voltage
+        else:
+            self.right_voltage = lambda x: right_voltage
+
+        self.phi = np.zeros(
+            (self.nx + 1 + 2*self.nxguardphi,
+            self.nz + 1 + 2*self.nzguardphi)
+        )
+
+        self.decompose_matrix()
+
+        print('Using direct solver.')
+        callbacks.installfieldsolver(self._run_solve)
+
+    def decompose_matrix(self):
+        """Function to build the superLU object used to solve the linear
+        system."""
+        self.nzsolve = self.nz + 1
+        self.nxsolve = self.nx + 3
```

Comment:
Why the +1 and +3?

Reply:
It creates the needed cells to enforce the specific boundary conditions. In the `z` direction we have Dirichlet boundaries so we only need one extra cell to fix the solution, while in the `x` direction we have periodic boundary conditions for which we need a ghost cell on each side and a third cell that is used as a dummy cell in order to enforce periodicity.

Reply:
Interesting, thanks